process/smartContract/processorV2/processV2.go

```diff
@@ -0,0 +1,2645 @@
+			}
+		}
+
+		scResults = append(scResults, result)
+	}
+
+	return createdAsyncCallBack, scResults
+}
+
+func (sc *scProcessor) useLastTransferAsAsyncCallBackWhenNeeded(
+	vmInput *vmcommon.VMInput,
+	outAcc *vmcommon.OutputAccount,
+	outputTransfer *vmcommon.OutputTransfer,
+	vmOutput *vmcommon.VMOutput,
+	tx data.TransactionHandler,
+	result *smartContractResult.SmartContractResult,
+	isCrossShard bool,
+) bool {
+	isAsyncTransferBackToSender := vmInput.CallType == vmData.AsynchronousCall &&
+		bytes.Equal(outAcc.Address, tx.GetSndAddr())
+	if !isAsyncTransferBackToSender {
+		return false
+	}
+
+	if !isCrossShard {
+		return false
+	}
+
+	if !sc.isTransferWithNoAdditionalData(outputTransfer.SenderAddress, outAcc.Address, outputTransfer.Data) {
+		return false
+	}
+
+	addReturnDataToSCR(vmOutput, result)
+	result.CallType = vmData.AsynchronousCallBack
+	result.GasLimit, _ = core.SafeAddUint64(result.GasLimit, vmOutput.GasRemaining)
+
+	var err error
+	result.Data, err = sc.prependAsyncParamsToData(
+		contexts.CreateCallbackAsyncParams(hooks.NewVMCryptoHook(), vmInput.AsyncArguments), result.Data)
+	if err != nil {
+		log.Debug("processed built in functions error (async params extraction)", "error", err.Error())
+		return false
+	}
+
+	return true
+}
+
+func (sc *scProcessor) prependAsyncParamsToData(asyncParams [][]byte, data []byte) ([]byte, error) {
+	function, args, err := sc.argsParser.ParseCallData(string(data))
+	if err != nil {
+		return nil, err
+	}
+
+	callData := txDataBuilder.NewBuilder()
+	callData.Func(function)
+
+	for _, arg := range args {
+		callData.Bytes(arg)
+	}
+
+	for _, asyncParam := range asyncParams {
+		callData.Bytes(asyncParam)
+	}
+
+	return callData.ToBytes(), nil
+}
+
+func (sc *scProcessor) getESDTParsedTransfers(sndAddr []byte, dstAddr []byte, data []byte,
+) (*vmcommon.ParsedESDTTransfers, error) {
+	function, args, err := sc.argsParser.ParseCallData(string(data))
+	if err != nil {
+		return nil, err
+	}
+
+	parsedTransfer, err := sc.esdtTransferParser.ParseESDTTransfers(sndAddr, dstAddr, function, args)
+	if err != nil {
+		return nil, err
+	}
+
+	return parsedTransfer, nil
+}
+
+func (sc *scProcessor) isTransferWithNoAdditionalData(sndAddr []byte, dstAddr []byte, data []byte) bool {
+	if len(data) == 0 {
+		return true
+	}
+
+	parsedTransfer, err := sc.getESDTParsedTransfers(sndAddr, dstAddr, data)
+	if err != nil {
+		return false
+	}
+
+	return len(parsedTransfer.CallFunction) == 0
+}
+
+// createSCRForSender(vmOutput, tx, txHash, acntSnd)
+// give back the user the unused gas money
+func (sc *scProcessor) createSCRForSenderAndRelayer(
+	vmOutput *vmcommon.VMOutput,
+	tx data.TransactionHandler,
+	txHash []byte,
+	callType vmData.CallType,
+) (*smartContractResult.SmartContractResult, *smartContractResult.SmartContractResult) {
+	if vmOutput.GasRefund == nil {
+		// TODO: compute gas refund with reduced gasPrice if we need to activate this
```

Comment:
make it big.NewInt(0) without the if - as it is not used.

Reply:
fixed